addons/mail/models/discuss/discuss_channel.py

```diff
@@ -467,18 +467,30 @@ def _sync_field_names(self):
     # ------------------------------------------------------------
 
     def _subscribe_users_automatically(self):
-        new_members = self._subscribe_users_automatically_get_members()
-        if new_members:
-            to_create = [
-                {'channel_id': channel_id, 'partner_id': partner_id}
-                for channel_id in new_members
-                for partner_id in new_members[channel_id]
-            ]
-            # sudo: discuss.channel.member - adding member of other users based on channel auto-subscribe
-            self.env['discuss.channel.member'].sudo().create(to_create)
-        for channel in self:
-            for group in channel.group_ids:
-                Store(bus_channel=group).add(channel, extra_fields={"is_pinned": True}).bus_send()
+        if not (new_members_to_create := self._subscribe_users_automatically_get_members()):
+            return
+        to_create = [
+            {"channel_id": channel_id, "partner_id": partner_id}
+            for channel_id in new_members_to_create
+            for partner_id in new_members_to_create[channel_id]
+        ]
+        # sudo: discuss.channel.member - adding member of other users based on channel auto-subscribe
+        new_members = self.env["discuss.channel.member"].sudo().create(to_create)
+        notifications = defaultdict(lambda: self.env["discuss.channel.member"])
+        for member in new_members:
+            bus_channel = member._bus_channel()
+            notifications[bus_channel] |= member
+        for bus_channel, members in notifications.items():
+            members = members.with_prefetch(new_members.ids)
+            channels = members.channel_id.with_prefetch(new_members.channel_id.ids)
```

Comment:
prefetch through relations is automatically assigned based on existing prefetch, I don't think adding it for channel is necessary